### ISSUE TITLE
3.3.0 Failing test: TestPluginRegistry.test_emptyPlugin

### ISSUE THREAD

**Issue by hroncok** [Ultimaker/Uranium#345]
This is what i get when running tests on Python 3.6, in Fedora's rawhide build environment:

```
=================================== FAILURES ===================================
_____________________ TestPluginRegistry.test_emptyPlugin ______________________

self = <TestPluginRegistry.TestPluginRegistry object at 0x7f12582d28d0>
registry = <TestPluginRegistry.FixtureRegistry object at 0x7f12582de828>

    def test_emptyPlugin(self, registry):
        registry.loadPlugin("EmptyPlugin")
        with pytest.raises(KeyError):
>           registry.getPluginObject("EmptyPlugin")

tests/PluginRegistry/TestPluginRegistry.py:72: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <TestPluginRegistry.FixtureRegistry object at 0x7f12582de828>
plugin_id = 'EmptyPlugin'

    def getPluginObject(self, plugin_id: str) -> PluginObject:
        if plugin_id not in self._plugins:
            self.loadPlugin(plugin_id)
        if plugin_id not in self._plugin_objects:
>           raise PluginNotFoundError(plugin_id)
E           UM.PluginError.PluginNotFoundError: Could not find plugin EmptyPlugin

UM/PluginRegistry.py:645: PluginNotFoundError
----------------------------- Captured stdout call -----------------------------
[MainThread] UM.PluginRegistry.loadPlugin [379]: Plugin EmptyPlugin did not return any objects to register
[MainThread] UM.PluginRegistry.loadPlugin [379]: Plugin EmptyPlugin did not return any objects to register
=============================== warnings summary ===============================
tests/benchmarks/conftest.py
  Skipping benchmarks because pytest-benchmark plugin was not found.

-- Docs: http://doc.pytest.org/en/latest/warnings.html
========= 1 failed, 260 passed, 26 skipped, 1 warnings in 8.34 seconds =========
```

Apparently, the test expects an KeyError, but PluginNotFoundError is risen instead.

This might be relevant commit https://github.com/Ultimaker/Uranium/commit/cdbb916096277c91394476e8fc35697fbc71e915  (or not) .

I'm not sure if "Could not find plugin EmptyPlugin" is the desired result and the tests are broken, or if it should find the plugin and later KeyError on the plugin being empty.

**Comment by Ghostkeeper** [comment]
I've been getting a segfault in Uranium's tests recently. Not at the same place in the tests every time though. It's random.

@diegopradogesto has been working on fixing these tests on the [WIP_improve_initialization](https://github.com/Ultimaker/Uranium/tree/WIP_improve_initialization) branch. Maybe that branch works better for you?

**Comment by hroncok** [comment]
Tests from that branch all pass.

**Comment by hroncok** [comment]
I believe this commit is the fixer, let em test that: 72e584d799450e7ede0667a8e875526bce21e5fd

**Comment by hroncok** [comment]
Actually just:

```patch
diff --git a/tests/PluginRegistry/TestPluginRegistry.py b/tests/PluginRegistry/TestPluginRegistry.py
index 0f8456e7c..43a8354e1 100644
--- a/tests/PluginRegistry/TestPluginRegistry.py
+++ b/tests/PluginRegistry/TestPluginRegistry.py
@@ -68,7 +72,7 @@ def test_disabledPlugin(self, registry):
 
     def test_emptyPlugin(self, registry):
         registry.loadPlugin("EmptyPlugin")
-        with pytest.raises(KeyError):
+        with pytest.raises(PluginNotFoundError):
             registry.getPluginObject("EmptyPlugin")
 
```

That's with the line of what I said in the report:

> Apparently, the test expects an KeyError, but PluginNotFoundError is risen instead.

**Comment by Ghostkeeper** [comment]
Indeed this is fixed with commit 72e584d799450e7ede0667a8e875526bce21e5fd.
